src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -276,18 +278,16 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         val selfArg = fn.selfParameter
 
         if (selfArg != null && superFn.selfParameter == null) {
-            holder.createErrorAnnotation(selfArg,
-                "Method `${fn.name}` has a `${selfArg.canonicalDecl}` declaration in the impl, but not in the trait [E0185]")
+            RsDiagnostic.DeclMissingFromTraitError(selfArg, fn, selfArg.canonicalDecl).addToHolder(holder)
         } else if (selfArg == null && superFn.selfParameter != null) {
-            holder.createErrorAnnotation(params,
-                "Method `${fn.name}` has a `${superFn.selfParameter?.canonicalDecl}` declaration in the trait, but not in the impl [E0186]")
+            RsDiagnostic.DeclMissingFromImplError(params, fn, "${superFn.selfParameter?.canonicalDecl}").addToHolder(holder)
```

Comment:
It's preffered to pass a concrete objects to `RsDiagnostic`, not their string representation